### PR TITLE
Add DisableLocalhostConnectionRateLimit configuration option

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

The current connection rate limit doesn't differentiate between connections that are coming from localhost vs connections that are coming from a remote machine. This means that the connection rate limit is unable to distinguish between multiple hosts running locally. This becomes a problem, since when running a local host network, one node can saturate the connection limits, and prevent other nodes from connecting to a pseudo relay.

To work around that, a flag was added `DisableLocalhostConnectionRateLimit` that would allow us to skip the connection limit for incoming connection whose IP address is the localhost. To retain backward compatibility, the flag is still there, and users can toggle it off.

## Test Plan

Update existing unit test; test manually against large local network.

### CODE REVIEW

File: network/requestTracker.go [hunk 1]
```diff
@@ -501,3 +507,13 @@ func (rt *RequestTracker) getForwardedConnectionAddress(header http.Header) (ip
 	}
 	return
 }
+
+// isLocalhost returns true if the given host is a localhost address.
+func isLocalhost(host string) bool {
+	for _, v := range []string{"localhost", "127.0.0.1", "[::1]", "::1"} {
```
Comment (brianolson) [inline network/requestTracker.go:513]: Go is also currently representing localhost as "[::]"
(this is what I see in a algod.net, `[::]:4160`)

**Review by brianolson — COMMENTED** [review]
I like it. I'm pretty sure there are times I would have used this in some dev testing.
I think we need at least one more string for 'localhost' `[::]`

File: network/requestTracker.go [hunk 2]
```diff
@@ -285,8 +285,11 @@ func (rt *RequestTracker) Accept() (conn net.Conn, err error) {
 		rt.hostRequests.pruneRequests(rateLimitingWindowStartTime)
 		originConnections := rt.hostRequests.countOriginConnections(trackerRequest.remoteHost, rateLimitingWindowStartTime)
 
+		remoteHostIsNonLocal := (!rt.config.DisableLocalhostConnectionRateLimit) || (!isLocalhost(trackerRequest.remoteHost))
```
Comment (algonautshant) [inline network/requestTracker.go:288]: I prefer naming this rateLimitedRemoteHost
This can be local host but reads remote Host Is Non Local.

**Review by algonautshant — APPROVED** [review]
Looks good.
It is a good idea to expand the bandwidth for local connections.

**Review by brianolson — DISMISSED** [review]

**Review by tsachizehub — APPROVED** [review]

**Pull request merged**